test/remote/gateways/remote_reach_test.rb

```diff
@@ -74,6 +75,60 @@ def test_failed_capture
     assert_equal 'Not Found', response.message
   end
 
+  #
+  # The following refund test was commented or it is optional because take
+  # ~1 minute to get a successful response
+  #
+
+  # def test_successful_refund
+  #   purchase = @gateway.purchase(@amount, @credit_card, @options)
+  #   sleep(1.minute)
+  #   response = @gateway.refund(@amount, purchase.authorization, { reference_id: 'REFUND_TAG' })
+
+  #   assert_success response
+  #   assert response.params['response'][:RefundId].present?
+  # end
+
+  def test_failed_refund
+    purchase = @gateway.purchase(@amount, @credit_card, @options)
+    response = @gateway.refund(@amount, purchase.authorization, { reference_id: 'REFUND_TAG' })
+
+    assert_failure response
+    assert_equal 'OrderStateInvalid', response.params['response'][:Error][:Code]
```

Comment:
⚠️  Warning:

If you are getting this as a real error code, please check and modify the `error_code_from` method, so you can check the error from response.error_code instead of checking the actual response body

Reply:
Done!